daemon/logger/journald/read.go

```diff
@@ -325,17 +293,26 @@ func (s *journald) readLogs(logWatcher *logger.LogWatcher, config logger.ReadCon
 		close(logWatcher.Msg)
 		return
 	}
+	if config.Follow {
+		// Initialize library inotify watches early
+		rc = C.sd_journal_get_fd(j)
+		if rc < 0 {
```

Comment:
> some of the old code using < C.int(0) instead of < 0; any idea if there was a reason for that?

This is because Go is strict wrt mixing different types, so say it doesn't allow comparison like `i != 2` in case `i` is `C.int` -- one needs to write it as `i != C.int(2)`. But it seems like in simpler cases like `< 0` comparisons Go is fine with mixing types.

Reply:
> if rc := C.sd_journal_get_fd(j); rc < 0 {

Yes, I prefer this form too, to limit var scope, thanks for suggestion! I have added a commit fixing this (together with a few other nitpicks).